### DISCUSSION TITLE
<title>I got stuck while making an Excel file in c#.

### DISCUSSION THREAD

**Discussion by whereismylife** [community/community#45764] [Discussions] [answered]
### Select Topic Area

Product Feedback

### Body

hi 

I made an Excel file using text boxes in the c# environment. It was possible to make it, but I made it possible only on my computer to save it. What I want is to be on Desktop on any computer.if you answer this, problem. I thank you for your kindness.

        private void button5_Click(object sender, EventArgs e)
        {
            string path = @"c:\Users\User1015\Desktop\data.csv";
            if (!File.Exists(path))
            {
                using (StreamWriter sw = File.CreateText(path))
                {
                    sw.WriteLine("Feret,minFeret,h (mm),minArea,maxArea,min V,max V");
                }

            }
            using (StreamWriter sw = File.AppendText(path))
            {
                sw.WriteLine($"{textBox1.Text},{textBox2.Text},{textBox6.Text},{textBox3.Text},{textBox4.Text},{textBox5.Text},{textBox7.Text}");
                sw.Close();
            }
        }

**Comment by jge162** [comment, marked as answer]
try changing the path -> `string path = AppDomain.CurrentDomain.BaseDirectory + @"data.csv";`

**Reply by whereismylife** [reply]
oh. it is very good answer, but i want to know easily installed in Desktop main screen. for example, C:\Users\jung hoe jun\Desktop like this in my computer. like this 
![image](https://user-images.githubusercontent.com/106086190/216809457-c5e02646-b92c-4d8c-a527-0569999104ce.png)

**Reply by jge162** [reply]
Can you please elaborate more what you want to accomplish, I do not understand your new request.

**Reply by whereismylife** [reply]
The code you told me will be automatically installed in the folder built on all computers, right?

**Reply by jge162** [reply]
yes

**Reply by whereismylife** [reply]
thank you so much!

**Reply by jge162** [reply]
no problem, if you are satisfied with answer please consider marking answered so you can help someone else out with solution. thank you gain for you post @whereismylife

**Comment by mohamed653** [comment]
well i used this nuget package  **EPPlus**  and worked fine with me but don't forget to install the version that match the dotnet version you are working with

here is the code  i used to export a datagridview into excel 


```
private void ExportToExcel(DataGridView dataGridView)
{
    using (ExcelPackage excelPackage = new ExcelPackage())
    {
        ExcelWorksheet worksheet = excelPackage.Workbook.Worksheets.Add("Sheet1");

        // Export DataGridView headers to Excel
        for (int i = 0; i < dataGridView.Columns.Count; i++)
        {
            worksheet.Cells[1, i + 1].Value = dataGridView.Columns[i].HeaderText;
        }

        // Export DataGridView rows to Excel
        for (int i = 0; i < dataGridView.Rows.Count; i++)
        {
            for (int j = 0; j < dataGridView.Columns.Count; j++)
            {
                worksheet.Cells[i + 2, j + 1].Value = dataGridView.Rows[i].Cells[j].Value?.ToString();
            }
        }

        // Save the Excel file
        SaveFileDialog saveFileDialog = new SaveFileDialog();
        saveFileDialog.Filter = "Excel Workbook (*.xlsx)|*.xlsx";
        if (saveFileDialog.ShowDialog() == DialogResult.OK)
        {
            FileInfo fileInfo = new FileInfo(saveFileDialog.FileName);
            excelPackage.SaveAs(fileInfo);
            MessageBox.Show("Exported to Excel successfully!", "Success", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }
    }
}
```
